webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/ComposeService.scala

```diff
@@ -0,0 +1,177 @@
+package com.scalableminds.webknossos.datastore.services
+
+import com.scalableminds.util.geometry.Vec3Double
+import com.scalableminds.util.io.PathUtils
+import com.scalableminds.util.tools.{Fox, FoxImplicits}
+import com.scalableminds.webknossos.datastore.dataformats.n5.{N5DataLayer, N5SegmentationLayer}
+import com.scalableminds.webknossos.datastore.dataformats.precomputed.{
+  PrecomputedDataLayer,
+  PrecomputedSegmentationLayer
+}
+import com.scalableminds.webknossos.datastore.dataformats.wkw.{WKWDataLayer, WKWSegmentationLayer}
+import com.scalableminds.webknossos.datastore.dataformats.zarr.{ZarrDataLayer, ZarrSegmentationLayer}
+import com.scalableminds.webknossos.datastore.dataformats.zarr3.{Zarr3DataLayer, Zarr3SegmentationLayer}
+import com.scalableminds.webknossos.datastore.models.datasource.{
+  CoordinateTransformation,
+  DataLayer,
+  DataSource,
+  DataSourceId,
+  GenericDataSource
+}
+import net.liftweb.util.Helpers.tryo
+import play.api.libs.json.{Json, OFormat}
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path}
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+
+case class ComposeRequest(
+    newDatasetName: String,
+    targetFolderId: String,
+    organizationName: String,
+    scale: Vec3Double,
+    layers: Seq[ComposeLayer]
+)
+
+object ComposeRequest {
+  implicit val composeRequestFormat: OFormat[ComposeRequest] = Json.format[ComposeRequest]
+}
+case class ComposeLayer(
```

Comment:
```suggestion
case class ComposeRequestLayer(
```

(Maybe this would make it clearer that this class is not an intermediate representation of a whole layer, but is just used in the request to identify one and specify what is to happen with it)